mm2src/coins/utxo.rs

```diff
@@ -1050,6 +1052,17 @@ impl<T: UtxoCommonOps> CoinAssocTypes for T {
     type Sig = Signature;
     type SigParseError = MmError<secp256k1::Error>;
 
+    fn my_addr(&self) -> &Self::Address {
+        match &self.as_ref().derivation_method {
+            DerivationMethod::SingleAddress(addr) => addr,
+            unimplemented => unimplemented!("{:?}", unimplemented),
+        }
+    }
+
+    fn parse_address(&self, address: &str) -> Result<Self::Address, Self::AddressParseError> {
+        Ok(Address::from_str(address)?)
+    }
```

Comment:
```suggestion
    fn parse_address(&self, address: &str) -> Result<Self::Address, Self::AddressParseError> {
        Address::from_str(address).map_to_mm(|e| e)
    }
```

Reply:
Both options actually do the same thing. Could you please provide more info on why `map_to_mm` might be preferred over original code?

Reply:
For me, it makes more sense when reading it. Constructing a `Result` and unwinding the inner `Result`, which has a `From` implementation on top of its Error. I don't know, I know they are essentially doing the same thing, but mapping the error without unwinding or touching (directly) the `Result` seems like a more elegant way to me. Feel free to keep it as it is if you think otherwise; maybe it's just me.

Reply:
Well, it is more of a personal preference for me too: using `map` and not performing a conversion on error in closure seems strange 🙂 Anyway, it's worth using a unified approach on such cases, so maybe @shamardy will help us to decide what we should use now and in the future? 🙂

Reply:
This is tricky since I prefer to avoid using Ok for returns whenever possible. However, as @artemii235 highlighted, not performing a conversion on errors in closures can seem odd. Therefore, I propose a third solution for such cases:
```rust
Address::from_str(address).map_err(MmError::from)
```
This shows that the error was converted which was implied implicitly by `?` while avoiding wrapping. The problem with this third approach is that we have to import the error type for errors other than `MmError`. So I suggest the following guideline:
- Try to avoid wrapping when possible. However, if avoiding wrapping leads to the explicit import of an error type,  then using `?` and wrapping in `Ok` is prefered.

Reply:
Great note, thanks! I pushed the code.

Reply:
I guess Address::from_str(address).map_to_mm(keys::Error::from) is also an option if we'd like to stick with our helpers.

BTW about the discussion above my 2 cents: I also noticed that map_to_mm is absolutely needed when we have to convert the whole result being returned. But when we unwrap results with '?' the standard map_err also works. I can see both options are used and maybe good to indicate a preferable option

Reply:
Yes, there are multiple places where we use `map_err(|_| MmError::new())` instead of `map_to_mm`, these were due to not knowing `map_to_mm` or forgetting about it. We should always prioritize `map_to_mm` when converting from `NotMmError` to `MmError`, same goes for `or_mm_err`.